### PR TITLE
Adds another planetstation mob, the "Seedling"

### PR DESCRIPTION
**Pull request by Ergovisavi**
Adds another planetstation mob, the "Seedling". It's basically a gun turret that looks like a flower. It shoots large volleys of energy bullets, and **has a laser attack that automatically hits you if you do not interrupt its attack by hitting it before it fires**. It won't start shooting at you from offscreen, you have to be onscreen to catch its attention first. If you go offscreen after it starts charging up and you know the laser is incoming, it will hit you.

Bring a friend to stab it in the butthole while you dodge its fire, or be mlg pro and dodge the shots and hit it between volleys

It looks like this:

seedling_sprites

It kills you like this:

flower_power_2

[
:cl:
add:  Adds the "seedling" planetstation mob to the backend
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 1]
```diff
@@ -578,11 +578,336 @@
...
+	owner.overlays -= beam_overlay
+	angle = target_angle
+	var/matrix/final = matrix()
+	final.Turn(target_angle)
+	beam_overlay.transform = final
+	owner.overlays += beam_overlay
+
+
+/datum/status_effect/seedling_beam_indicator/be_replaced()
+	owner.underlays -= beam_overlay
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: underlays?

Comment (Ergovisavi) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: I am not a clever man

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 2]
```diff
@@ -578,11 +578,336 @@
...
+			return
+	AttackRecovery()
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/Volley()
+	if(combatant_state == SEEDLING_STATE_WARMUP && target)
+		combatant_state = SEEDLING_STATE_ACTIVE
+		update_icons()
+		var/datum/callback/cb = CALLBACK(src, .proc/InaccurateShot)
+		var/i
+		for(i=0, i<13, i++)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: for(var/i in 1 to however_many_shots_this_is)

**Comment by Armhulen** [comment]
now, i know it's an immobile creature but is this player controlled?

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 3]
```diff
@@ -578,11 +578,336 @@
...
+	damage = 10
+	damage_type = BURN
+	light_range = 2
+	flag = "energy"
+	light_color = LIGHT_COLOR_YELLOW
+	hitsound = 'sound/weapons/sear.ogg'
+	hitsound_wall = 'sound/weapons/effects/searwall.ogg'
+	nondirectional_sprite = TRUE
+
+/obj/item/projectile/seedling/Bump(atom/A, yes)//Stops seedlings from destroying other jungle mobs through FF
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: This is ```Collide``` now.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 4]
```diff
@@ -578,11 +578,336 @@
...
+	tick()
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/Destroy()
+	if(owner)
+		owner.overlays -= beam_overlay
+	QDEL_NULL(beam_overlay)
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/tick()
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: holy shit the appearance churn is real.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 5]
```diff
@@ -578,11 +578,336 @@
...
+	var/mob/living/simple_animal/hostile/jungle/seedling/our_seedling
+
+/obj/seedling_weakpoint/Destroy()
+	. = ..()
+	if(our_seedling && !QDELETED(our_seedling))
+		if(!our_seedling.stat)
+			our_seedling.combatant_state = SEEDLING_STATE_NEUTRAL
+			our_seedling.death()
+
+/obj/seedling_weakpoint/proc/RandomizeScreenLocation()
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: ??

Is the icon state the same sprite but in multiple locations? if so, just randomise ```pixel_x```/```pixel_y```

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 4 above)
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: also the indicator looks ugly anyway is this really necessary?
If it is, remove it for now, when 512 comes along there's a way to do this in a cheaper manner, and you can have it then.

Comment (Ergovisavi) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: If the seedling is offscreen, or you are engaged with two separate seedlings, or two seedlings are charging up at two separate people, there'd be no way to tell which one you need to smack to free yourself without an indicator.

I do think the churn is too high and so I'd like to use a cheaper method but this information needs to exist for the player

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: then, considering planestation will come way after 512, if ever, remove this for now and when the time comes I'll help you redo it with 512.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 5 above)
Comment (Ergovisavi) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: Icon isn't the same. It was originally a little dot on a random pixel_x/pixel_y but it looked kinda shit so I just made it in three separate locations with a specific sprite in each location

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: then rename everything.
```PickVisuals()```, ```RandomSprite()``` etc.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 6]
```diff
@@ -578,11 +578,336 @@
...
+		if(get_dist(src,living_target) > 7)
+			playsound(living_target,'sound/effects/seedling_chargeup.ogg', 100, 0)
+		solar_beam_identifier = world.time
+		addtimer(CALLBACK(src, .proc/Beamu, living_target, solar_beam_identifier), 35)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/Beamu(mob/living/living_target, beam_id = 0)
+	if(combatant_state == SEEDLING_STATE_ACTIVE && living_target && beam_id == solar_beam_identifier)
+		if(living_target.z == z)
+			update_icons()
+			solar_beam = new(src,target,time=5,beam_icon_state="solar_beam",maxdistance = INFINITY,btype=/obj/effect/ebeam/solarbeam,beam_sleep_time = 0.5)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: I recommend making this from the turf to the target, since beam handles larger icon files weirdly.
You can see how in the gif.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 4 above)
Comment (DoctoreMad) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: It can laser people offscreen?
WHY?

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 7]
```diff
@@ -578,11 +578,336 @@
 	icon_state = "mook_leap_cloud"
 	layer = BELOW_MOB_LAYER
 	pixel_x = -16
-	pixel_y = -16
+	pixel_y = -12
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: What's with this mook change?

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 8]
```diff
@@ -578,11 +578,336 @@
...
+	ranged = TRUE
+	ranged_cooldown_time = 10
+	projectiletype = /obj/item/projectile/seedling
+	projectilesound = 'sound/weapons/pierce.ogg'
+	robust_searching = TRUE
+	stat_attack = UNCONSCIOUS
+	anchored = TRUE
+	var/combatant_state = SEEDLING_STATE_NEUTRAL
+	var/obj/seedling_weakpoint/weak_point
+	var/datum/beam/solar_beam = null
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: remove the ` = null` please

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 3 above)
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: /obj/item/projectile/seedling/Collide(atom/A)

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 9]
```diff
@@ -578,11 +578,336 @@
...
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/tick()
+	var/target_angle = Get_Angle(owner, target)
+	owner.overlays -= beam_overlay
+	angle = target_angle
+	var/matrix/final = matrix()
+	final.Turn(target_angle)
+	beam_overlay.transform = final
+	owner.overlays += beam_overlay
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: add_overlay

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 10]
```diff
@@ -578,11 +578,336 @@
...
+
+/datum/status_effect/seedling_beam_indicator/Destroy()
+	if(owner)
+		owner.overlays -= beam_overlay
+	QDEL_NULL(beam_overlay)
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/tick()
+	var/target_angle = Get_Angle(owner, target)
+	owner.overlays -= beam_overlay
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: cut_overlay

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 11]
```diff
@@ -578,11 +578,336 @@
...
+/datum/status_effect/seedling_beam_indicator/on_apply()
+	beam_overlay = mutable_appearance('icons/mob/jungle/arachnid.dmi', "seedling_beam_indicator")
+	beam_overlay.pixel_x = -16 -owner.pixel_x
+	beam_overlay.pixel_y = -16 -owner.pixel_y
+	tick()
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/Destroy()
+	if(owner)
+		owner.overlays -= beam_overlay
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: cut_overlay

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 1 above)
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: cut_overlay

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 12]
```diff
@@ -578,11 +578,336 @@
...
+			playsound(living_target,'sound/effects/seedling_chargeup.ogg', 100, 0)
+		solar_beam_identifier = world.time
+		addtimer(CALLBACK(src, .proc/Beamu, living_target, solar_beam_identifier), 35)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/Beamu(mob/living/living_target, beam_id = 0)
+	if(combatant_state == SEEDLING_STATE_ACTIVE && living_target && beam_id == solar_beam_identifier)
+		if(living_target.z == z)
+			update_icons()
+			solar_beam = new(src,target,time=5,beam_icon_state="solar_beam",maxdistance = INFINITY,btype=/obj/effect/ebeam/solarbeam,beam_sleep_time = 0.5)
+			INVOKE_ASYNC(solar_beam, /datum/beam.proc/Start)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: we have the atom/Beam proc use that instead

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 13]
```diff
@@ -578,11 +578,336 @@
...
+		combatant_state = SEEDLING_STATE_ACTIVE
+		update_icons()
+		var/datum/callback/cb = CALLBACK(src, .proc/InaccurateShot)
+		var/i
+		for(i=0, i<13, i++)
+			addtimer(cb, i)
+		addtimer(CALLBACK(src, .proc/AttackRecovery), 14)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/InaccurateShot()
+	if(target && !QDELETED(target) && combatant_state == SEEDLING_STATE_ACTIVE)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: if(!QDELETED(target) && combatant_state == SEEDLING_STATE_ACTIVE)

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 14]
```diff
@@ -578,11 +578,336 @@
...
+			return
+		var/turf/our_turf = get_turf(src)
+		var/obj/item/projectile/seedling/readied_shot = new /obj/item/projectile/seedling(our_turf)
+		readied_shot.current = our_turf
+		readied_shot.starting = our_turf
+		readied_shot.firer = src
+		readied_shot.original = target
+		readied_shot.yo = target.y - our_turf.y + rand(-1,1)
+		readied_shot.xo = target.x - our_turf.x + rand(-1,1)
+		readied_shot.fire()
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: Don't you have to call prepare_pixel_projectile or something? @kevinz000 ?

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 15]
```diff
@@ -578,11 +578,336 @@
...
+/mob/living/simple_animal/hostile/jungle/seedling/proc/AttackRecovery()
+	if(combatant_state == SEEDLING_STATE_ACTIVE)
+		combatant_state = SEEDLING_STATE_RECOVERY
+		update_icons()
+		ranged_cooldown = world.time + ranged_cooldown_time
+		if(target)
+			face_atom(target)
+		if(solar_beam)
+			qdel(solar_beam)
+			solar_beam = null
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: QDEL_NULL(solar_beam)

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 16]
```diff
@@ -578,11 +578,336 @@
...
+	update_icons()
+	walk(src,0)
+	wander = FALSE
+	addtimer(CALLBACK(src, .proc/SeedlingRegenerate), SEEDLING_REGENERATION_TIME)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/SeedlingRegenerate()
+	if(weak_point && !QDELETED(weak_point) && combatant_state == SEEDLING_STATE_REGENERATING)
+		revive(TRUE)
+		mouse_opacity = 1
+		weak_point.loc = src
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: forceMove

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 17]
```diff
@@ -578,11 +578,336 @@
...
+	icon = 'icons/mob/jungle/arachnid.dmi'
+	icon_state = "seedling_weakpoint1"
+	pixel_x = -16
+	pixel_y = -14
+	layer = ABOVE_ALL_MOB_LAYER
+	var/mob/living/simple_animal/hostile/jungle/seedling/our_seedling
+
+/obj/seedling_weakpoint/Destroy()
+	. = ..()
+	if(our_seedling && !QDELETED(our_seedling))
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: just `!QDELETED(our_seedling)`

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 18]
```diff
@@ -578,11 +578,336 @@
...
+		if(!our_seedling.stat)
+			our_seedling.combatant_state = SEEDLING_STATE_NEUTRAL
+			our_seedling.death()
+
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: our_seedling = null

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm [hunk 19]
```diff
@@ -578,11 +578,336 @@
...
+		to_chat(user, "<span class='danger'>You need something sharp for this!</span>")
+		return
+	RandomizeScreenLocation()
+	return ..()
+
+/obj/seedling_weakpoint/play_attack_sound()
+	playsound(src, 'sound/weapons/bladeslice.ogg', 50, 1)
+
+/mob/living/simple_animal/hostile/jungle/seedling/update_stat()
+	if(weak_point && !QDELETED(weak_point))
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: !QDELETED(weak_point)

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 18 above)
Comment (Ergovisavi) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: Why would I need to null out our_seedling? The object deletes itself as soon as this finishes.

**Comment by Tacolizard** [comment]
this seems kind of op

**Comment by Armhulen** [comment]
leapers are only strong alone and these are also the case

**Comment by DoctoreMad** [comment]
Come on mate, have you tried fighting it yourself or are you going off looks?
I could say it looks easy to me and it would be nonconstructive as fuck because I'm going off a .gif

**Comment by DoctoreMad** [comment]
Tbh this could be a boss if it was surrounded by rglass in a titanium wall dome so the projectiles kept reflecting everywhere.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 18 above)
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: ergo, it will *TRY* to qdel itself after this finishes, btu since our_seedling is still pointing to the mob, there's a reference, and it won't qdel properly, so it'll sit in the qdel queue for a little while and THEN it'll trigger a harddel (a ```del()``` which is stupidly bad)

Mobs already don't qdel well, but I think simple animals can, so don't make it worse.

**Comment by KingPhilipIII** [comment]
Alright. I'll be honest. I don't know what planetstation even is. Can someone please enlighten me.

**Comment by ChangelingRain** [comment]
I think the weak point/regen thing isn't necessary, maybe just have it explode briefly after killing it so you don't waste that sprite

**Comment by Ergovisavi** [comment]
Yeah I think at this point I'll just cut the weakpoint thing, it was just a spitballing idea and I don't think it actually has much merit, since it doesn't really change how you act against the mob unless you were shooting at it.

**Comment by DoctoreMad** [comment]
@KingPhilipIII ONE DAY A BUNCH OF IDEA GUYS THOUGHT IT WOULD BE COOL IF THE CREW COULD GO AROUND CONQUERING PLANETS IN THE NAME OF NANOTRASEN. ALSO THEY THOUGHT THAT IT WOULD ALSO BE PRETTY COOL IF THERE WERE MORE ENEMY MOBS WITH ADVANCED AI.
AND SO "PLANETSTATION" WAS FORMED!

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 14 above)
Comment (Ergovisavi) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: I dunno, the volley seems to work perfectly fine and to my expectations.

**Comment by Ergovisavi** [comment]
Removed the weakpoint gimmick it's just not something viable as a gameplay mechanic due to the way our combat works and is more suited for a game with a better aim system

**Comment by DoctoreMad** [comment]
That reminds me.
@KorPhaeron About your goal of having a planet-naming scroll on the planetstation project list, isn't it already done?

**Comment by KorPhaeron** [comment]
Yeah

**Comment by Cheridan** [comment]
An enemy that requires you juke waves of bullets in a game where any damage makes you go slower sounds like its not going to be very fun

**Comment by PraiseRatvar** [comment]
the projectile sprite looks bad especially when it is firing multiple at once. 
perhaps give it a completely different projectile instead of "solar energy" or make the projectile sprite smaller and/or make it fire less projectiles

**Comment by Ergovisavi** [comment]
@Cheridan The environment in which these are fought are assumed to have dense trees to use as cover, so if you're not confident in your ability to dodge volleys, use cover. Or bring a friend. Also, you have to be shot by four bullets before you suffer any slowdown, so getting clipped a little won't significantly hinder you.

@PraiseRatvar You're going to have to be more specific other than "looks bad"

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 14 above)
Comment (kevinz000) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: il care about the bug when i get dunked by it and i make a i ded pr within 0.1 seconds of death @Cyberboss 
shitposting aside I dunno it broke without it for robustin's gun turrets but I do believe it's only needed if a PLAYER is shooting it...

Comment (kevinz000) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: however i will say using this and not randomized angles results in less random :P

**Comment by kevinz000** [comment]
>making ebeam fake-hitscans
>>but i already made a beam rifle projectile that hitscans properly and has pixel and angle accurate sprites!!!

sigh.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 14 above)
Comment (Ergovisavi) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: I don't want actual randomization on the projectile volley. All the shots should be crossing within one tile of the target.

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: @kevinz000 do you even read code?

**Comment by Ergovisavi** [comment]
@kevinz000 Your beam rifle fires a projectile and can hit targets on the way to the actual target which is explicitly the behavior I do not want

It's not a true laser, all the visual effects are just set dressing.

File: code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm (hunk 14 above)
Comment (kevinz000) [inline code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm]: @RemieRichards heh

**Comment by kevinz000** [comment]
@Ergovisavi ah ok.

**Comment by RemieRichards** [comment]
Ergo refuses to wait for 512 where his visual effect will be cheap enough to not cause painful client lag, so I am closing this until 512 is stable at which point we can reopen this PR or create a new one.

**Comment by Armhulen** [comment]
>:(

**Comment by ChangelingRain** [comment]
Okay so the pointing to the seedling thing isn't viable until byond 512 and ergo doesn't want to add the mob without the laser beam and pointing thing, so, until Byond 512...

**Comment by Ergovisavi** [comment]
Guess this is on the backburner until after 512 without Kor or someone else stepping in, RIP

**Comment by Tokiko1** [comment]
Mind elaborating what so expensive about the visual effect is?

**Comment by RemieRichards** [comment]
Every status effect ```tick()``` ergo is creating a new appearance for the visual effect (rotating it X degrees, as the effect is a compass), and then applying that effect to the target mob, that's 2 appearances per tick where with 512 it can be just 1 per tick, once 512 is also stable we can look into porting something like /vg/'s hook system so that the compass won't ever update unless either the seedling or the target move (which is the 100% best perfect case, meaning you have nothing happening on ```tick()``` which keeps things nice and cheap)

Creating and replacing appearances in DM is horrendously slow, it's why 510/11 introduced ```/mutable_appearance```, every change to an appearance var creates a new one, eg:
```
bob.pixel_y = 32 //new appearance
bob.pixel_x = 32 //another new appearance
bob.color = "red" //another new appearance
```
That's 3 appearance, 2 of which didn't need to exist (the pixel_x/y ones only existed for a short time before being destroyed and replaced by the next one along)
With 510/11 and ```/mutable_appearance```s something like the above can instead be:
```
var/mutable_appearance/MA = new()
MA.pixel_y = 32
MA.pixel_x = 32
MA.color = "red"
bob.appearance = MA //This 'commits' MA, turning it from a /mutable_appearance into a real appearance, meaning only 1 was created per n-appearance variables changed, instead of n per n
```
With 512, there's ```visual_contents```, which is like ```overlays``` and ```underlays```, but the objects inside ```visual_contents``` REMAIN as objects, they don't get flattened into appearances, meaning any update to their appearance doesn't need to go through the whole overlay/underlay painful sorting process.

So tl;dr 512 will be one step closer to perfect, then a hook-system can make it 100% perfect.

**Comment by Ergovisavi** [comment]
Changed to using a screen object, which should serve roughly the same purpose without the need to constantly tweak overlays

**Comment by Armhulen** [comment]
ok it's been open for two days lets merge it up guys

**Comment by optimumtact** [comment]
@RemieRichards did you review the latest version?

**Comment by Ergovisavi** [comment]
Can I get some traction on this?

**Comment by RemieRichards** [comment]
Really? assigning me just so I can say "Yes it's fine"?

"Yes it's fine"

**Comment by Cyberboss** [comment]
:^)
